### PR TITLE
Clean up Kubernetes PullPolicy

### PR DESCRIPTION
**Pull request by dchen1107**
With this PR, kubernetes wounldn't change the user's setting. Also when PullPolicy is absent, the default is PullIfNotPresent except the case of image has :latest tag, and default is set to be PullAlways.

Also removed unused methods on determining PullPolicy at the run time from helper.go. 

cc @bgrant0607

### CODE REVIEW

**Comment by dchen1107** [comment]
cc @smarterclayton This is to address your issue in one of PRs on PullPolicy.

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -266,6 +266,28 @@ func validateLifecycle(lifecycle *api.Lifecycle) errs.ValidationErrorList {
...
 }
 
+// TODO(dchen1107): Move this along with other defaulting values
+func validatePullPolicyWithDefault(ctr *api.Container) errs.ValidationErrorList {
+	allErrors := errs.ValidationErrorList{}
+
+	if len(ctr.ImagePullPolicy) == 0 {
+		parts := strings.SplitN(ctr.Image, ":", 2)
+		// Check image tag
+		if len(parts) == 2 && parts[1] == "latest" {
```
Comment (brendandburns) [inline pkg/api/validation/validation.go]: what about images that specify a registry?  e.g. foo.bar.com:5000/my/image:latest

**Comment by brendandburns** [comment]
This basically LGTM except with the one issue parsing image names.

File: pkg/api/validation/validation.go (hunk 1 above)
Comment (satnam6502) [inline pkg/api/validation/validation.go]: Re:
```
parts := strings.SplitN(ctr.Image, ":", 2)
```
Surely nothing good can happen if there is more than one colon in the image name? Perhaps it's worth raising an error for this case?

**Comment by dchen1107** [comment]
@brendandburns I addressed your comment, and add your example as a testcase in validation_test.go. PTAL?

**Comment by bgrant0607** [comment]
We should create a standard library routine for parsing image names. Another instance of that is here:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/contrib/podex/podex.go#L133

and I copied it for some of my config generators.

**Comment by smarterclayton** [comment]
We've got one i'd be happy if were upstream.

https://github.com/openshift/origin/blob/master/pkg/image/api/helper.go

File: pkg/api/validation/validation.go [hunk 2]
```diff
@@ -266,6 +266,28 @@ func validateLifecycle(lifecycle *api.Lifecycle) errs.ValidationErrorList {
...
+	if len(ctr.ImagePullPolicy) == 0 {
+		parts := strings.Split(ctr.Image, ":")
+		// Check image tag
+		if parts[len(parts)-1] == "latest" {
+			ctr.ImagePullPolicy = api.PullAlways
+		} else {
+			ctr.ImagePullPolicy = api.PullIfNotPresent
+		}
+	}
+	if ctr.ImagePullPolicy != api.PullAlways &&
```
Comment (thockin) [inline pkg/api/validation/validation.go]: cleaner as switch() with default being the error?

**Comment by thockin** [comment]
Clayton's routine looks good (as a followup), but I might beg it to return a struct instead of 4 values..

**Comment by smarterclayton** [comment]
It's not quite perfect - I got irritated at the magic assumptions so I had it set "library" (the real default namespace) and "latest" (since upstream has indicated that is a docker cli convention only).  I can take a follow on to normalize it for whatever use we want.

**Comment by dchen1107** [comment]
@bgrant0607 I added TODO, and will cleanup the code paths related ParseImageName in a separate PR.

File: pkg/api/validation/validation.go (hunk 2 above)
Comment (thockin) [inline pkg/api/validation/validation.go]: poking this comment

Comment (dchen1107) [inline pkg/api/validation/validation.go]: Ahh, missed this one. Will update this.

**Comment by bgrant0607** [comment]
I'm guessing that Kubelet doesn't invoke this code for pods created via other means, such as from a file, but that doesn't need to be handled in this PR. 

LGTM

**Pull request merged**